### PR TITLE
Web: Add chart package with stat vis components [Line Chart]

### PR DESCRIPTION
**Pull request by vovakulikov**
## Context

This PR introduces a new shared package for static visualization components `/charts`. The problem at the moment is we have a few separate implementations of the charts (admin statistic page, batch changes burndown chart, code stats pie chart, insight charts) which is not good obviously in terms of the codebase and different designs/UI/UX aspects. 
This package should provide all needed charts components and by this solve problems with inconsistencies in charts codebases/designs.

In particular, this PR adds just the `<LineChart />` chart as a first component. Since we have this component in many places it makes sense to start with it.

## Tech Details

In code insights charts we've been facing a lot of problems with customization of `@visx/xychart` primitives (like tooltip position, or data shape manipulation, hover chart state, ...etc). Because of this in new charts, we don't use this high-level abstraction anymore. Instead of this, we use more low-level blocks and primitives such as `@visx/shape`, `@visx/glyph` ... etc. 

For chart tooltip positioning in this PR, we use [floating UI package](https://floating-ui.com). In this PR we created a special separate component for chart tooltip only. But it might be a part of the wildcard component as a primitive and would be useful I believe in all components that have floating/popover parts (comboboxes, menu buttons, dropdowns, selects...) 

## Next step

- Bar and bar-stack charts
- Pie chart
- Adopting these charts in different consumers and removing in place chart implementations.
- Update `@visx` packages versions 

## For reviewers 

Note this PR has a larger size than average PR. I tried to divide this PR into separate PRs but it turned out it would be a lot of small and sequence PRs (I've found that this sequence PRs approach doesn't work really well when we work on something in one place like one separated component) Instead of multi PRs approach I divided changes into small commits

- Floating component @sourcegraph/frontend-platform https://github.com/sourcegraph/sourcegraph/pull/29372/commits/a9547c127714181f5c6c453ac32d3188e289b2b8

### CODE REVIEW

**Comment by Joelkw** [comment]
Silly question @vovakulikov  – does this make it easier if we want to enable "you can add batch changes burndown charts to an insights dashboard"?

**Comment by vovakulikov** [comment]
@Joelkw Well this particular PR doesn't make it easier directly but this simplifies Line Chart logic a lot and extracts the tooltip component as a shared part (which is good because we could reuse it for burn chart as well). 

Eventually, I want to migrate all our charts in Sourcegraph on charts that Code Insights uses (and have one base for all static visualization in our product). So the answer here would be - yes it makes it easier but this is just a first step for all chart migration where we could reuse different charts in a different places (like batch changes burn chart in code insights pages).

File: client/web/src/charts/components/line-chart/components/axis/Tick.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Tick component displays tick label for each axis line of chart.
+ */
+export const Tick: React.FunctionComponent<TickRendererProps> = props => {
+    const { formattedValue, ...tickLabelProps } = props
+
+    // Hack with Group + Text (aria hidden)
+    // Because the Text component renders text inside of svg element and text element with tspan
+    // this makes another nested group for a11y tree. To avoid "group - end group"
+    // phrase in voice over we hide nested children from a11y tree and put explicit aria-label
+    // on the parent Group element with role text
```
Comment (unclejustin) [inline client/web/src/charts/components/line-chart/components/axis/Tick.tsx:31]: Thanks for this explanation and for thinking of our users! 🙌

File: client/web/src/charts/components/line-chart/utils/ticks.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    // Generate max density ticks (d3 scale generation)
+    const ticks = getTicks(scale) as number[]
+
+    if (ticks.length <= 2) {
+        return ticks
+    }
+
+    // Calculate desirable number of ticks
+    const numberTicks = Math.max(2, Math.floor(space / pixelsPerTick))
```
Comment (unclejustin) [inline client/web/src/charts/components/line-chart/utils/ticks.ts:64]: question: non-blocking

Do the `2`s here need to be the same? In other words, does it make sense to extract this number to a constant?

**Review by unclejustin — APPROVED** [review]

**Comment by unclejustin** [comment]
@vovakulikov LGTM 🌮  Just one small question.

**Pull request merged**